cmd/create.go

```diff
@@ -102,18 +108,29 @@ func printAdvancedInstructionsMessage(deplDir string) {
 	logging.Info(modulewriter.InstructionsPath(deplDir))
 }
 
-func expandOrDie(path string) config.DeploymentConfig {
+func expandOrDie(path string, dPath string) config.DeploymentConfig {
 	dc, ctx, err := config.NewDeploymentConfig(path)
 	if err != nil {
 		logging.Fatal(renderError(err, ctx))
 	}
-	// Set properties from CLI
-	if err := setCLIVariables(&dc.Config, cliVariables); err != nil {
+
+	var ds config.DeploymentSettings
+	var dCtx config.YamlCtx
+	if dPath != "" {
+		ds, dCtx, err = config.NewDeploymentSettings(dPath)
+		if err != nil {
+			logging.Fatal(renderError(err, dCtx))
+		}
+	}
+	if err := setCLIVariables(&ds, cliVariables); err != nil {
```

Comment:
How I am reading this is that CLI vars supersedes deployment file, deployment file supersedes blueprint. Can you confirm? If so then I am happy.

Reply:
Yes. The conceptual intent is that command line variables are now the highest precedence deployment variables.

The technical intent is to implement a well defined behavior that aligns with the status quo: command line options always take precedence. It might be worth a reminder that `--vars` can be invoked multiple times, with later invocations at higher priority. e.g.

```
ghpc create -d test.yaml --vars "project_id=yads" --vars "project_id=bla"
```

and

```
ghpc create --vars "project_id=yada" --vars "project_id=bla" -d test.yaml 
```

both result in `vars.project_id="bla"` irrespective of the contents of the deployment file.